beets/importer.py

```diff
@@ -547,6 +547,23 @@ def cleanup(self, copy=False, delete=False, move=False):
 
         items = self.imported_items()
 
+        # If any original album art files were reused, leave them untouched.
+        for item in items:
+            album = item.get_album()
+            if album:
+                self.old_art_paths.discard(album.artpath)
+
+        # Delete remaining original album art files.
+        for old_art_path in self.old_art_paths:
+            log.debug('Deleting orphaned album art: {0}', old_art_path)
+            util.remove(syspath(old_art_path), True)
+            # XXX: Ideally the directory would get pruned with self.prune()
+            # below, but when calling `beet import -L`, self.toppath is set
+            # to None, making self.prune() a no-op; hence this workaround of
+            # calling util.prune_dirs() directly.
+            util.prune_dirs(os.path.dirname(old_art_path),
+                            clutter=config['clutter'].as_str_seq())
```

Comment:
That is, unless pruning for `import -L` works fine except for album art and this is just needed to keep the current functionality intact.